frontend/src/core/l10n.js

```diff
@@ -13,7 +13,31 @@ define(['require', 'jquery', 'polyglot', 'core/origin'], function(require, $, Po
   /**
   * Initialise from language file
   */
-  var locale = localStorage.getItem('lang') || 'en';
+
+  //Rudimentary language detection https://stackoverflow.com/questions/673905/best-way-to-determine-users-locale-within-browser
+  var locale;
+  if (typeof navigator !== 'undefined'){
```

Comment:
I'm not happy with this code and I think we may be better allowing the user to explicitly request a language or looking at the locale headers passed in the api request.